lib/OpenQA/Schema/ResultSet/JobSettings.pm

```diff
@@ -35,30 +35,21 @@ Given a perl hash, will create a ResultSet of job_settings
 sub query_for_settings {
     my ($self, $args) = @_;
 
-    my @joins;
     my @conds;
     # Search into the following job_settings
     for my $setting (keys %$args) {
-        if ($args->{$setting}) {
-            # for dynamic self joins we need to be creative ;(
-            my $tname = 'me';
-            if (@conds) {
-                $tname = "siblings";
-                if (@joins) {
-                    $tname = "siblings_" . (int(@joins) + 1);
-                }
-                push(@joins, 'siblings');
-            }
-            my $setting_value = ($args->{$setting} =~ /^:\w+:/) ? {'like', "$&%"} : $args->{$setting};
-            push(
-                @conds,
-                {
-                    "$tname.key"   => $setting,
-                    "$tname.value" => $setting_value
-                });
-        }
+        next unless $args->{$setting};
+        # for dynamic self joins we need to be creative ;(
+        my $tname         = 'me';
+        my $setting_value = ($args->{$setting} =~ /^:\w+:/) ? {'like', "$&%"} : $args->{$setting};
+        push(
+            @conds,
+            {
+                "$tname.key"   => $setting,
+                "$tname.value" => $setting_value
+            });
     }
-    return $self->search({-and => \@conds}, {join => \@joins});
+    return $self->search({-and => \@conds});
```

Comment:
Looks like the only place the parameters for joining can be passed is the API route for cancelling jobs by settings if the user provides these parameters. So I *guess* we can delete the code at this point.